### PR TITLE
feat: adding support for frameworks greater than and equal to NET6

### PR DESCRIPTION
**Pull request by andygjp**
Fixes https://github.com/OData/AspNetCoreOData/issues/1294

### CODE REVIEW

**Comment by andygjp** [comment]
@microsoft-github-policy-service agree

File: src/Microsoft.AspNetCore.OData/Common/TypeHelper.cs [hunk 1]
```diff
@@ -93,7 +93,7 @@ public static bool IsDateTime(Type clrType)
             return Type.GetTypeCode(underlyingTypeOrSelf) == TypeCode.DateTime;
         }
 
-#if NET6_0
+#if NET6_0_OR_GREATER
```
Comment (ElizabethOkerio) [inline src/Microsoft.AspNetCore.OData/Common/TypeHelper.cs:96]: I think we need to get rid of the conditionals completely.

Comment (andygjp) [reply src/Microsoft.AspNetCore.OData/Common/TypeHelper.cs:96]: I can do that

Comment (andygjp) [reply src/Microsoft.AspNetCore.OData/Common/TypeHelper.cs]: Done

**Review by gathogojr — APPROVED** [review]

**Review by WanjohiSammy — APPROVED** [review]

**Pull request merged**